### ISSUE TITLE
E:\Users\Administrator\.pm2\logs    but i  find the similar path at c:\Users\Administrator\.pm2\logs 

### ISSUE THREAD

**Issue by Messilimeng** [Unitech/pm2#1616]
PM2:     at Domain.<anonymous> (C:\Users\Administrator\AppData\Roaming\npm\node_modules\pm2\lib\Satan.js:651:13)
PM2:     at Domain.run (domain.js:191:14)
PM2:     at Object.<anonymous> (C:\Users\Administrator\AppData\Roaming\npm\node_modules\pm2\lib\Satan.js:650:7)
PM2:     at Module._compile (module.js:434:26)
PM2:     at Object.Module._extensions..js (module.js:452:10)
PM2: 2015-09-16 09:34:05: [[[[ PM2/God daemon launched ]]]]
PM2: 2015-09-16 09:34:05: BUS system [READY] on port \\.\pipe\pub.sock
PM2: 2015-09-16 09:34:05: RPC interface [READY] on port \\.\pipe\rpc.sock
PM2: 2015-09-16 09:34:05: Starting execution sequence in -fork mode- for app name:app id:0
PM2: 2015-09-16 09:34:05: Trace: [Error: can not create directories (logs/pids):ENOENT: no such file or directory, mkdir 'E:\Users\Administrator\.pm2\logs']
PM2:     at Object.God.logAndGenerateError (C:\Users\Administrator\AppData\Roaming\npm\node_modules\pm2\lib\God\Methods.js:30:15)
PM2:     at C:\Users\Administrator\AppData\Roaming\npm\node_modules\pm2\lib\God\ForkMode.js:86:13
PM2:     at Object.module.exports.startLogging (C:\Users\Administrator\AppData\Roaming\npm\node_modules\pm2\lib\Utility.js:93:14)
PM2:     at Object.forkMode (C:\Users\Administrator\AppData\Roaming\npm\node_modules\pm2\lib\God\ForkMode.js:84:13)
PM2:     at Object.executeApp (C:\Users\Administrator\AppData\Roaming\npm\node_modules\pm2\lib\God.js:197:9)

**Comment by Unitech** [comment]
PM2 version?

Be more precise in the issue

thanks

**Comment by Messilimeng** [comment]
![qq 20150923152139](https://cloud.githubusercontent.com/assets/12272166/10039653/d38b1ae6-6206-11e5-84ee-837c32051dce.png)

**Comment by Messilimeng** [comment]
@Unitech  it is the laster version in windows.

**Comment by DavidBenko** [comment]
+1

**Comment by Messilimeng** [comment]
when I  use pm2 kill   than  I can not start the app.

**Comment by pokemane** [comment]
Chiming in with the same problem.

```
2015-10-21 18:37:47: Starting execution sequence in -fork mode- for app name:test id:5
2015-10-21 18:37:47: Trace: [Error: can not create directories (logs/pids):ENOENT: no such file or directory, mkdir 'E:\Users\USER\.pm2\logs']
    at Object.God.logAndGenerateError (C:\Users\USER\AppData\Roaming\npm\node_modules\pm2\lib\God\Methods.js:30:15)
    at C:\Users\USER\AppData\Roaming\npm\node_modules\pm2\lib\God\ForkMode.js:86:13
    at Object.module.exports.startLogging (C:\Users\USER\AppData\Roaming\npm\node_modules\pm2\lib\Utility.js:93:14)
    at Object.forkMode (C:\Users\USER\AppData\Roaming\npm\node_modules\pm2\lib\God\ForkMode.js:84:13)
    at Object.executeApp (C:\Users\USER\AppData\Roaming\npm\node_modules\pm2\lib\God.js:197:9)
    at ex (C:\Users\USER\AppData\Roaming\npm\node_modules\pm2\lib\God.js:367:18)
    at prepare (C:\Users\USER\AppData\Roaming\npm\node_modules\pm2\lib\God.js:373:7)
    at Server.onmessage (C:\Users\USER\AppData\Roaming\npm\node_modules\pm2\node_modules\pm2-axon-rpc\lib\server.js:105:6)
    at emitTwo (events.js:87:13)
    at RepSocket.emit (events.js:172:7)
```

**Comment by iredjee** [comment]
+1 same issue.
- Windows 8,
- Node v4.2.1
- npm v3.3.9

Logs
```
2015-10-28 15:05:11: Starting execution sequence in -fork mode- for app name:index id:3
2015-10-28 15:05:11: Trace: [Error: can not create directories (logs/pids):ENOENT: no such file or directory, mkdir 'D:\Users\USER\.pm2\logs']
    at Object.God.logAndGenerateError (C:\Users\USER\AppData\Roaming\npm\node_modules\pm2\lib\God\Methods.js:30:15)
    at C:\Users\USER\AppData\Roaming\npm\node_modules\pm2\lib\God\ForkMode.js:86:13
    at Object.module.exports.startLogging (C:\Users\USER\AppData\Roaming\npm\node_modules\pm2\lib\Utility.js:93:14)
    at Object.forkMode (C:\Users\USER\AppData\Roaming\npm\node_modules\pm2\lib\God\ForkMode.js:84:13)
    at Object.executeApp (C:\Users\USER\AppData\Roaming\npm\node_modules\pm2\lib\God.js:197:9)
    at ex (C:\Users\USER\AppData\Roaming\npm\node_modules\pm2\lib\God.js:367:18)
    at prepare (C:\Users\USER\AppData\Roaming\npm\node_modules\pm2\lib\God.js:373:7)
    at Server.onmessage (C:\Users\USER\AppData\Roaming\npm\node_modules\pm2\node_modules\pm2-axon-rpc\lib\server.js:105:6)
    at emitTwo (events.js:87:13)
    at RepSocket.emit (events.js:172:7)
```
I guess there is a directory problem, look at the log line
```
mkdir 'D:\Users\USER\.pm2\logs'
```
It uses disk D, i.e. the disk where the actual code located, but for sure it must be disk C.

Waiting for the fix :)

**Comment by hmny** [comment]
Same problem:
```
PM2 Starting execution sequence in -fork mode- for app name:www id:10
PM2 Trace: [Error: can not create directories (logs/pids):ENOENT: no such file or directory, mkdir 'D:\Users\username\.pm2\logs']
PM2     at Object.God.logAndGenerateError (C:\Users\username\AppData\Roaming\npm\node_modules\pm2\lib\God\Methods.js:45:15)
PM2     at C:\Users\username\AppData\Roaming\npm\node_modules\pm2\lib\God\ForkMode.js:86:13
PM2     at Object.module.exports.startLogging (C:\Users\username\AppData\Roaming\npm\node_modules\pm2\lib\Utility.js:93:14)
PM2     at Object.forkMode (C:\Users\username\AppData\Roaming\npm\node_modules\pm2\lib\God\ForkMode.js:84:13)
PM2     at Object.executeApp (C:\Users\username\AppData\Roaming\npm\node_modules\pm2\lib\God.js:197:9)
PM2     at ex (C:\Users\username\AppData\Roaming\npm\node_modules\pm2\lib\God.js:368:18)
PM2     at prepare (C:\Users\username\AppData\Roaming\npm\node_modules\pm2\lib\God.js:374:7)
PM2     at Server.onmessage (C:\Users\username\AppData\Roaming\npm\node_modules\pm2\node_modules\pm2-axon-rpc\lib\server.js:105:6)
PM2     at emitTwo (events.js:87:13)
PM2     at RepSocket.emit (events.js:172:7)
```

**Comment by ReklatsMasters** [comment]
my temporary solution:

```sh
mkdir D:\Users\%USERNAME%\.pm2\logs
rmdir C:\Users\%USERNAME%\.pm2\logs
mklink /D C:\Users\%USERNAME%\.pm2\logs D:\Users\%USERNAME%\.pm2\logs
```

**Comment by alex-alfaro** [comment]
+1 Same issue here 
Windows 10
Node 4.2.3
npm 2.14.7
pm2 0.15.10
```
From the Log:
2015-12-08 11:22:29: Trace: [Error: can not create directories (logs/pids):ENOENT: no such file or directory, mkdir 'D:\Users\Alejandro\.pm2\logs']
    at Object.God.logAndGenerateError (C:\Users\Alejandro\AppData\Roaming\npm\node_modules\pm2\lib\God\Methods.js:45:15)
    at C:\Users\Alejandro\AppData\Roaming\npm\node_modules\pm2\lib\God\ForkMode.js:86:13
    at Object.module.exports.startLogging (C:\Users\Alejandro\AppData\Roaming\npm\node_modules\pm2\lib\Utility.js:93:14)
    at Object.forkMode (C:\Users\Alejandro\AppData\Roaming\npm\node_modules\pm2\lib\God\ForkMode.js:84:13)
    at Object.executeApp (C:\Users\Alejandro\AppData\Roaming\npm\node_modules\pm2\lib\God.js:197:9)
    at ex (C:\Users\Alejandro\AppData\Roaming\npm\node_modules\pm2\lib\God.js:368:18)
    at prepare (C:\Users\Alejandro\AppData\Roaming\npm\node_modules\pm2\lib\God.js:374:7)
    at Server.onmessage (C:\Users\Alejandro\AppData\Roaming\npm\node_modules\pm2\node_modules\pm2-axon-rpc\lib\server.js:105:6)
    at emitTwo (events.js:87:13)
    at RepSocket.emit (events.js:172:7)
```
I tried setting up the directories and links as @ReklatsMasters  suggested but still not getting it to work (no matter if running as Admin), this is the output:
```
D:\Dev\Complementors\MaiPenRai>pm2 start server.js --no-daemon --no-autorestart
pm2 launched in no-daemon mode (you can add DEBUG="*" env variable to get more m
essages)
undefined
[PM2] restartProcessId process id 3
[PM2] Process successfully started
┌──────────┬────┬──────┬──────┬────────┬─────────┬────────┬────────────┬────────
──┐
│ App name │ id │ mode │ pid  │ status │ restart │ uptime │ memory     │ watchin
g │
├──────────┼────┼──────┼──────┼────────┼─────────┼────────┼────────────┼────────
──┤
│ server   │ 3  │ fork │ 8184 │ online │ 0       │ 0s     │ 9.418 MB   │ disable
d │
└──────────┴────┴──────┴──────┴────────┴─────────┴────────┴────────────┴────────
──┘
 Use `pm2 show <id|name>` to get more details about an app
--no-daemon option enabled = do not exit pm2 CLI
fs.js:549
  return binding.open(pathModule._makeLong(path), stringToFlags(flags), mode);
                 ^

Error: ENOENT: no such file or directory, open 'C:\Users\Alejandro\.pm2\pm2.pid'

    at Error (native)
    at Object.fs.openSync (fs.js:549:18)
    at Object.fs.readFileSync (fs.js:397:15)
    at C:\Users\Alejandro\AppData\Roaming\npm\node_modules\pm2\lib\CLI.js:2260:4
2
    at C:\Users\Alejandro\AppData\Roaming\npm\node_modules\pm2\node_modules\pm2-
axon-rpc\lib\client.js:45:10
    at Parser.<anonymous> (C:\Users\Alejandro\AppData\Roaming\npm\node_modules\p
m2\node_modules\pm2-axon\lib\sockets\req.js:67:8)
    at emitOne (events.js:77:13)
    at Parser.emit (events.js:169:7)
    at Parser._write (C:\Users\Alejandro\AppData\Roaming\npm\node_modules\pm2\no
de_modules\pm2-axon\node_modules\amp\lib\stream.js:91:16)
    at doWrite (_stream_writable.js:292:12)
```

**Comment by hmny** [comment]
I solved this problem by re-installing the package:
```
npm uninstall -g pm2
npm install -g pm2
```

**Comment by Unitech** [comment]
I do not get it, logs folders should be created 

https://github.com/Unitech/pm2/blob/master/lib/CLI.js#L40

**Comment by hmny** [comment]
@Unitech 
It creates the logs folder, but look for it in a directory that doesn't exist. This problem happened to me when I followed the first suggestion [here](https://github.com/Unitech/pm2/issues/1079). I did made a script like 
```set HOMEDRIVE=D:
set HOMEPATH=\Users\%USERNAME%
```
and since then it complained that you don't have the `.logs` folder under your `D:` drive (I had it on `C:`). Reinstalling solved the problem anyway.
